src/transformers/modeling_tf_utils.py

```diff
@@ -253,60 +253,49 @@ def detect_tf_missing_unexpected_layers(model, resolved_archive_file):
     unexpected_layers = []
 
     with h5py.File(resolved_archive_file, "r") as f:
-        saved_layer_names = set(hdf5_format.load_attributes_from_hdf5_group(f, "layer_names"))
-        model_layer_names = set(layer.name for layer in model.layers)
-        missing_layers = list(model_layer_names - saved_layer_names)
-        unexpected_layers = list(saved_layer_names - model_layer_names)
+        saved_h5_model_layers_name = set(hdf5_format.load_attributes_from_hdf5_group(f, "layer_names"))
+        model_layers_name_value = {}
```

Comment:
This line and the for loop after should be replaced by
```
model_layers_name_value = {layer.name: layer for layer in model.layers}
```
This is easier code to read IMO and will be faster (dict comprehension is faster than a manual loop).